skills/repo.go

```diff
@@ -302,10 +302,68 @@ func (r *Repository) Grant(skillSnapName, skillName, slotSnapName, slotName stri
 }
 
 // Revoke revokes the named skill from the slot of the given snap.
+//
+// Revoke has three modes of operation that depend on the passed arguments:
+//
+// If all the arguments are specified revoke finds a specific skill slot and a
+// specific skill and revoke that skill from that skill slot. It is an error if
+// skill or skill slot cannot be found or if the grant does not exist.
+//
+// If skillSnapName and skillName are empty then revoke finds the specified
+// skill slot and revokes all the skills granted there. It is not an error if
+// there are no such skills but it is still an error if the skill slot does not
+// exist.
+//
+// If skillSnapName, skillName and slotName are all empty then revoke finds the
+// specified snap (designated by slotSnapName) and revokes all the skills from
+// all the skill slots found therein. It is not an error if there are no such
+// skills but it is still an error if the snap does not exist or has no slots
+// at all.
 func (r *Repository) Revoke(skillSnapName, skillName, slotSnapName, slotName string) error {
 	r.m.Lock()
 	defer r.m.Unlock()
 
+	switch {
```

Comment:
Do we prefer `if {...} else if {...} else {...}` or a `switch` when dealing with purely boolean conditions?